SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -1601,3 +1601,43 @@ void USpatialNetDriver::HandleOnConnectionFailed(const FString& Reason)
 	UE_LOG(LogSpatialOSNetDriver, Error, TEXT("Could not connect to SpatialOS. Reason: %s"), *Reason);
 	OnConnectionFailed.Broadcast(Reason);
 }
+
+uint32_t USpatialNetDriver::AddOpCallback(Worker_ComponentId ComponentId, const UserOpCallback& Callback)
+{
+	SpatialConstants::MIN_EXTERNAL_SCHEMA_ID <= ComponentId && ComponentId <= SpatialConstants::MAX_EXTERNAL_SCHEMA_ID;
+	uint32_t NewCallbackId = NextCallbackId++;
+	ComponentToCallbackIdMap.FindOrAdd(ComponentId).Add(NewCallbackId);
+	CallbackIdToDataMap.Add(NewCallbackId, UserOpCallbackData{ComponentId, Callback});
+	return NewCallbackId;
+}
+
+void USpatialNetDriver::RemoveOpCallback(uint32_t CallbackId)
```

Comment:
NIT- for stuff in the unreal interface we should probably use unreal types. I think Unreal has a typedef for for uint32_t as uint32